bin/gene_library_diversity.r

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env Rscript
+
+# read counts matrix
+input_file <- commandArgs(trailingOnly = TRUE)[1]
+count_matrix <- read.table(input_file, header = TRUE, stringsAsFactors = FALSE, sep = '\t')
+
+column_names <- colnames(count_matrix)
+column_indices <- 3:ncol(count_matrix)
+
+# empty data frame to store the results
+result_table <- data.frame(Sample = character(),
+                           'number_of_genes_contributing_to_1%_of_reads' = numeric(),
+                           'number_of_genes_contributing_to_5%_of_reads' = numeric(),
+                           'number_of_genes_contributing_to_10%_of_reads' = numeric(),
+                           'number_of_genes_contributing_to_50%_of_reads' = numeric(),
+                           'number_of_genes_contributing_to_80%_of_reads' = numeric(),
+                           stringsAsFactors = FALSE, check.names = FALSE)
+
+percentages <- c(1, 5, 10, 50, 80)
+
+# iterate over each sample column
+for (col_index in column_indices) {
+  col_name <- column_names[col_index]
+  col_values <- count_matrix[, col_index]
+
+  # calculate
+  genes_count <- sapply(percentages, function(percent) {
+    threshold <- percent / 100 * sum(col_values)
+    length(which(cumsum(col_values) <= threshold)) + 1
```

Comment:
This is wrong if the values are not **sorted in decreasing order**. Example:
```
10
10
2000
100
100
880
```
Total is 3000. 80% is 2400. If you compare the cumulative sum without ordering first,
```
10
20
2020
2120
2220 <= threshold up to this one
3100
```
you would find 6 genes contributing to 80%, while in fact there are only two genes contributing to 80%. The correct cumulative sum would be
```
2000 <= threshold up to this one
2880
2980
3080
3090
3100
```
The cumulative sum will always be monotonic, but its values will depend on the ordering of the values of the input vector.

Reply:
Understood! Thank you for your revision, I overlooked the fact that the values needed to be in decreasing order, I just wrongly assumed that they would. Your suggestion to add the ordering on line 26 will work but I'll wait for others to revise to commit the code to this PR